TypeSystem/flow-analysis/reachability_conditional_A02_t01.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Conditional statement: If `N` is a conditional statement of the
+/// form `if (E) S1 else S2` then:
+/// - Let `before(E) = before(N)`.
+/// - Let `before(S1) = split(true(E))`.
+/// - Let `before(S2) = split(false(E))`.
+/// - Let `after(N) = merge(after(S1), after(S2))`.
+///
+/// @description Checks that if `E` has the form `if-case` then even if all
+/// conditions are `true` then assignment in `then` branch is not treated as
+/// definite assignment.
```

Comment:
This library and the following 2 whose description says "even if" are basically focused on the fact that if-case (and pattern matching in general) makes no attempt to conclude at compile time that the outcome will be a specific boolean value, even in the case where the scrutinee is a boolean literal and the pattern is `== true` or `== false`. 

In other words, flow analysis considers the scrutinee to be "some boolean" even though it's specifically one of the literals `true` or `false`, and hence it doesn't conclude that "we will choose one specific branch, and the other one is unreachable". Instead, it considers both branches to be reachable.

So how do we say that? (I think there must be a better way to say it than "all conditions are `true`").

```suggestion
/// @description Checks that if `E` has the form `if-case` then the flow
/// analysis will not conclude that a specific branch will definitely be taken,
/// even in the case where the scrutinee is a boolean literal and the pattern
/// is of the form `== b` where `b` is also a boolean literal. This is tested
/// via the definite assignment status of a variable.
```

Reply:
Thank you. Updated here and below.